src/test/java/jenkins/branch/CustomOrganizationFolderDescriptorTest.java

```diff
@@ -76,35 +76,41 @@ public void removeIntegrationTestMocks() throws Exception {
     
     @Test
     public void noNavigatorNoFactoryInstalled() throws Exception {
-        assertEquals(1, ExtensionList.lookup(SCMNavigatorDescriptor.class).size());
-        assertEquals(SingleSCMNavigator.DescriptorImpl.class, ExtensionList.lookup(SCMNavigatorDescriptor.class).get(0).getClass());
-        assertEquals(0, ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class).size());
+        assertThat(ExtensionList.lookup(SCMNavigatorDescriptor.class),
+                hasItem(ExtensionList.lookupSingleton(SingleSCMNavigator.DescriptorImpl.class)));
+        // Ensure none of our mock factories sneaked into this test (though other MultiBranchProjectFactory impls may exist on the classpath)
+        assertThat(ExtensionList.lookup(OrganizationFolderTest.MockFactoryDescriptor.class), is(empty()));
```

Comment:
This is incomplete. There are two kinds of extensions here: `MultiBranchProjectFactoryDescriptor` (repo-level) and `SCMNavigatorDescriptor` (org-level). The original test is asserting the contents of both extension lists. Here you are asserting only one. Should do negative checks on both `MockFactoryDescriptor` and `MockNavigatorDescriptor`.